### PR TITLE
Fixes a typo and improves CIFAR10 example.

### PR DESCRIPTION
**Pull request by conor-f**
Tiny typo fix ("an loss function" to "a loss function")

### CODE REVIEW

**Comment by conor-f** [comment]
Added using top_k_categorical_accuracy to CIFAR10 CNN example and fixed it breaking when data_augmentation is set to False.

File: docs/templates/metrics.md [hunk 1]
```diff
@@ -17,7 +17,7 @@ model.compile(loss='mean_squared_error',
               metrics=[metrics.mae, metrics.categorical_accuracy])
 ```
 
-A metric function is similar to an [loss function](/losses), except that the results from evaluating a metric are not used when training the model.
+A metric function is similar to a [loss function](/losses), except that the results from evaluating a metric are not used when training the model.
```
Comment (fchollet) [inline docs/templates/metrics.md]: Right!

File: examples/cifar10_cnn.py [hunk 2]
```diff
@@ -27,6 +28,12 @@
 save_dir = os.path.join(os.getcwd(), 'saved_models')
 model_name = 'keras_cifar10_trained_model.h5'
 
+
+def top_3_acc(y_true, y_pred):
+    '''Example of how to use custom metrics - Shows accuracy when looking at
+    the top 3 predictions.'''
+    return top_k_categorical_accuracy(y_true, y_pred, k=3)
```
Comment (fchollet) [inline examples/cifar10_cnn.py]: I don't think this is useful in this example. Please remove.

File: examples/cifar10_cnn.py [hunk 3]
```diff
@@ -76,11 +83,7 @@
 
 if not data_augmentation:
     print('Not using data augmentation.')
-    model.fit(x_train, y_train,
-              batch_size=batch_size,
-              epochs=epochs,
-              validation_data=(x_test, y_test),
-              shuffle=True)
+    datagen = ImageDataGenerator()
```
Comment (fchollet) [inline examples/cifar10_cnn.py]: Please revert this change.

File: examples/cifar10_cnn.py (hunk 2 above)
Comment (conor-f) [reply examples/cifar10_cnn.py]: Sure

File: examples/cifar10_cnn.py (hunk 3 above)
Comment (conor-f) [reply examples/cifar10_cnn.py]: Reverting this means if data_augmentation is False that the example crashes. Will I revert this change with the top_k_accuracy and make a new commit just fixing the broken example?

Comment (fchollet) [reply examples/cifar10_cnn.py]: The example currently works fine with `data_augmentation=False`...

Comment (conor-f) [reply examples/cifar10_cnn.py]: When running with `data_augmentation=False`:

> Traceback (most recent call last):
  File "cifar10_cnn.py", line 132, in <module>
    evaluation = model.evaluate_generator(datagen.flow(x_test, y_test,
NameError: name 'datagen' is not defined`

[Instantiation line](https://github.com/fchollet/keras/blob/master/examples/cifar10_cnn.py#L87)
[Broken line](https://github.com/fchollet/keras/blob/master/examples/cifar10_cnn.py#L132)

Comment (fchollet) [reply examples/cifar10_cnn.py]: That's a bug indeed, but of a different kind. Test/evaluation shouldn't be run with data augmentation. Instead we should be using the Numpy array data + `predict`/`evaluate`.

Comment (fchollet) [reply examples/cifar10_cnn.py]: I removed this incorrect part of the code in the latest commit. Thanks for pointing it out!

Comment (conor-f) [reply examples/cifar10_cnn.py]: Cool :) Will close this PR so.
